include/xtensor-io/xhighfive.hpp

```diff
@@ -0,0 +1,666 @@
+/***************************************************************************
+* Copyright (c) 2016, Wolf Vollprecht, Sylvain Corlay and Johan Mabille    *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_IO_XHIGHFIVE_HPP
+#define XTENSOR_IO_XHIGHFIVE_HPP
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <highfive/H5DataSet.hpp>
+#include <highfive/H5DataType.hpp>
+#include <highfive/H5DataSpace.hpp>
+#include <highfive/H5File.hpp>
+
+#include <xtensor/xarray.hpp>
+#include <xtensor/xtensor.hpp>
+
+#include "xtensor-io.hpp"
+
+namespace xt
+{
+    namespace extensions
+    {
+        /**
+        * Check if a path exists (is a Group or DataSet) in an open HDF5 file.
+        *
+        * @param file opened HighFive::File
+        * @param path path of the Group/DataSet
+        */
+        inline bool exist(const HighFive::File &file, const std::string &path)
+        {
+            // find first "/"
+            std::size_t idx = path.find("/");
+
+            // loop over all groups
+            while (true)
+            {
+                // terminate if all "/" have been found
+                if (std::string::npos == idx)
+                {
+                    break;
+                }
+
+                // create group if needed
+                if (idx > 0)
+                {
+                    // get group name
+                    std::string name(path.substr(0,idx));
+                    // create if needed
+                    if (!file.exist(name))
+                    {
+                        return false;
+                    }
+                }
+
+                // proceed to next "/"
+                idx = path.find("/",idx+1);
+            }
+
+            return file.exist(path);
+        }
+
+        /**
+        * Recursively create groups in an open HDF5 file such that a DataSet can be created.
+        * For example if the path = "/path/to/dataset", this function will create the groups
+        * "/path" and "/path/to".
+        *
+        * @param file opened HighFive::File
+        * @param path path of the DataSet
+        */
+        inline void createGroup(HighFive::File &file, const std::string &path)
```

Comment:
Using camel case goes against our conventions (I guess this is HighFive inspired, right?) - but is this one of the functions you've pushed upstream as well? If so, do we expect to remove it in the future, once HighFive releases your additions?

If we expect to remove it, we should probably mention that in the comments to make it clear that this is an internal function.

If we don't want to remove it I think we should stick to snake_case.

Reply:
I will just use `create_group` for now. We can then always see what we want to do when things get implemented upstream.